avocado/utils/asset.py

```diff
@@ -171,6 +172,7 @@ def fetch(self):
             for item in self.locations:
                 urls.append(item)
 
+        cache_relative_dir = self._get_relative_dir(parsed_url)
```

Comment:
The implementation before was getting a cache directory for each url. Now it's getting only one cache dir for all urls. Why?

Reply:
Because it was wrong before :wink:

The point is that the controlling aspect of the behavior, and thus of the resulting cache directory, should be dependent on the *main* URL, as it's given in the `name` parameter.  The various sources (the URLs) are just locations the asset can be obtained from.

Reply:
BTW, the tests in this module can become a lot better and simpler if/when we do something like this: https://trello.com/c/CKP7YS6G/1481-on-cache-check-for-asset-fetcher